cmd/test/main.go

```diff
@@ -472,7 +477,25 @@ func test(ctx *TestCommandContext, paths []string, testCommandData *TestCommandD
 		CliVersion:            ctx.CliVersion,
 		IsCI:                  ctx.CiContext.IsCI,
 		Quiet:                 testCommandData.Quiet,
-	})
+	}
+	err = evaluation.PrintResults(evaluationData)
+
+	if testCommandData.SaveResults != "" {
+		if evaluation.IsWritableDirectory(testCommandData.SaveResults) {
```

Comment:
This function looks a little redundant to me, I mean why not just try to `writeFile` and if an error is returned then just return the error?

The error returned from this should look like this, which is understandable enough, WDYT?
@alonzyl 
![image](https://github.com/datreeio/datree/assets/51760613/01c8cb97-20da-4652-9b91-7200bee0d487)